contributors/design-proposals/grow-volume-size.md

```diff
@@ -0,0 +1,159 @@
+# Growing Persistent Volume size
+
+## Goals
+
+Enable users to increase size of PVs that their pods are using. The user will update PVC for requesting a new size. Underneath we expect that - a controller will apply the change to PV which is bound to the PVC.
+
+## Non Goals
+
+* Reducing size of Persistent Volumes: We realize that, reducing size of PV is way riskier than increasing it. Reducing size of a PV could be a destructive operation and it requires support from underlying file system and volume type. In most cases it also requires that file system being resized is unmounted.
+
+* Rebinding PV and PVC: Kubernetes will only attempt to resize the currently bound PV and PVC and will not attempt to relocate data  from a PV to a new PV and rebind the PVC to newly created PV.
+
+## Use Cases
+
+* As a user I am running Mysql on a 100GB volume - but I am running out of space, I should be able to increase size of volume mysql is using without losing all my data. (*online and with data*)
+* As a user I created a PVC requesting 2GB space. I am yet to start a pod with this PVC but I realize that I probably need more space. Without having to create a new PVC, I should be able to request more size with same PVC. (*offline and no data on disk*)
+* As a user I was running a rails application with 5GB of assets PVC. I have taken my application offline for maintenance but I would like to grow asset PVC to 10GB in size. (*offline but with data*)
+
+## Volume Plugin Matrix
+
+
+| Volume Plugin   | Supports Resize   | Requires File system Resize | Supported in 1.7 Release |
+| ----------------| :---------------: | :--------------------------:| :----------------------: |
+| EBS             | Yes               | Yes                         | Yes                      |
+| GCE PD          | Yes               | Yes                         | Yes                      |
+| Azure Disk      | Yes               | Yes                         | No                       |
+| Cinder          | Yes               | Yes                         | Yes                      |
+| Vsphere         | Yes               | Yes                         | No                       |
+| Ceph RBD        | Yes               | Yes                         | No                       |
+| Host Path       | No                | No                          | No                       |
+| GlusterFS       | Yes               | No                          | Yes                      |
+| Azure File      | No                | No                          | No                       |
+| Cephfs          | No                | No                          | No                       |
+| NFS             | No                | No                          | No                       |
```

Comment:
As long as - necessary fields/annotation that exist on PV that tells kubelet that a fs resize is needed - Kubelet will perform fs resize.  Ideally - we don't want admins to modify those annotations/fields manually - but we are not going to put additional restrictions if they want to!

So in a sense, if admin has resized a volume manually (via ebs resize command) and did not also update the PV object with new size and `volume.alpha.kubernetes.io/fs-resize-pending` annotation then no FS resize will be performed on kubelet. But if admin manually edits the PV and makes same changes that resize controller would do, then we can't distinguish manual changes from resize controller change and hence a FS resize will be performed.  But just because you can, doesn't mean you should.